tests/zfs-tests/tests/functional/zvol/zvol_misc/zvol_misc_snapdev.ksh

```diff
@@ -148,6 +148,12 @@ log_must zfs set snapdev=hidden $ZVOL
 check_missing $SNAPDEV
 log_must zfs set snapdev=hidden $ZVOL
 check_missing $SNAPDEV
+# 2.4 Verify we don't hit a race condition setting snapdev in rapid succession
```

Comment:
One potential explanation for this is that udev utilizes a number of child process in parallel to run the various rules.  That makes it possible that without an explicit `udevadm settle` in between they can run out of order, particularly in a case like this.